__tests__/index.test.ts

```diff
@@ -0,0 +1,68 @@
+import request from 'supertest';
+import express from 'express';
+
+const app = express();
+
+import { generateCronExpression, generateReadableExpression } from '../src/cronGenerator';
+
+app.get('/v1/generate', (req, res) => {
+  const cronExpression = generateCronExpression();
+
+  let readableExpression: string;
+  try {
+    readableExpression = generateReadableExpression(cronExpression);
+  } catch (error: any) {
+    console.error(`Error generating readable expression: ${error.message}`);
+    readableExpression = 'Invalid expression';
+  }
+
+  res.json({
+    expression: cronExpression,
+    readableExpression: readableExpression,
+  });
+});
+
+describe('GET /v1/generate', () => {
+  it('should return a predefined readableExpression for a known cronExpression', async () => {
+    const knownCronExpression = '22 4 * *';
+    const knownReadableExpression = 'Every minute, between 4:22 AM and 4:22 AM, on every day';
+
+    jest.spyOn(require('../src/cronGenerator'), 'generateCronExpression').mockReturnValue(knownCronExpression);
```

Comment:
Instead of mocking, can we instead seed the random function? That'd be more clean and you'd actually have more test coverage.

Reply:
Closing this as seeding is not possible in JS.